### DISCUSSION TITLE
select component with server side search

### DISCUSSION THREAD

**Discussion by aparasa1777** [twilio-labs/paste#2984] [Q&A] [answered]
Hi team,

I'm looking to see if there are any design/UI component options (which work for legacy code in numbers-console-frontend) for our use case below

We have a Modal which invokes an API to fetch a list of addresses (with pageSize = 1500), and enables search on "Friendly Name" attribute of the addresses. However, there are 2 issues we're trying to address

- For users with >1500 addresses, there is no way for the user to pick an address which isn't part of initial API response
- Could we make the select box lazy load (load 50 records on load, and load more on scroll) and leverage API driven search

![image](https://user-images.githubusercontent.com/104798478/214708576-aae7d8bf-0180-4f43-8ab5-4451ec6e3d8d.png)

Would appreciate any thoughts. Thank You!

**Comment by gloriliale** [comment, marked as answer]
Hi @aparasa1777! It sounds like the best component for your use case would be our [Combobox component](https://paste.twilio.design/components/combobox). Althought Combobox does not currently include lazy loading, we do have [a ticket in our backlog to add this functionality](https://issues.corp.twilio.com/browse/DSYS-3530). The best way to make sure we are able to prioritize that feature in Q2 would be to to submit a [dependency request](https://airtable.com/shrvMvL3BG6Zm0f9l) with us. 

In the meantime, other teams that have built a combobox with async functionality; you can check out these discussions https://github.com/twilio-labs/paste/discussions/1762 and https://github.com/twilio-labs/paste/discussions/1208 for more context. Hope this helps, let me know!
